ddprof-lib/src/main/cpp/flightRecorder.h

```diff
@@ -127,9 +128,13 @@ class Recording {
   static char *_java_command;
 
   RecordingBuffer _buf[CONCURRENCY_LEVEL];
+  // we have several sets to avoid lock contention
+  // we have a second dimension to allow a switch in the active set
+  std::unordered_set<int> _thread_ids[CONCURRENCY_LEVEL][2];
+  std::atomic<int> _active_index{0};  // 0 or 1 globally
```

Comment:
I wonder - can this be factored out, maybe to `ThreadFilter`? 
It feels kind of out of place here.